docs/_source/guides/llms/practical_guides/fine_tune.md

```diff
@@ -1,24 +1,305 @@
+After [collecting the responses](/guides/llms/practical_guides/collect_responses.html) from our `FeedbackDataset` we can start fine-tuning our LLMs and other models. Due to the customizability of the task, this might require setting up a custom post-processing workflow but we will provide some good toy examples for the [LLM approaches](/guides/llms/conceptual_guides/rlhf.html): pre-training, supervised fine-tuning, and reinforcement learning through human feedback (RLHF). However, we also still provide for other NLP tasks like text classification.
+## The `ArgillaTrainer`
 
-## Supervised finetuning
+The `ArgillaTrainer` is a wrapper around many of our favorite NLP libraries. It provides a very intuitive abstract representation to facilitate simple training workflows using decent default pre-set configurations without having to worry about any data transformations from Argilla.
 
-The goal of Supervised Fine Tuning (SFT) is to optimize this pre-trained model to generate the responses that users are looking for. After pre-training a causal language model, it can generate feasible human text, but it will not be able to have proper `answers` to `question` phrases posed by the user in a conversational or instruction set. Therefore, we need to collect and curate data tailored to this use case to teach the model to mimic this data. We have a section in our docs about [collecting data for this task](../conceptual_guides/sft.html) and there are many good [pre-trained causal language models](https://huggingface.co/models?pipeline_tag=text-generation&sort=downloads) available on Hugging Face.
+Using the `ArgillaTrainer` is straighforward but does slightly differ per task.
+
+1. First, we then define a `TaskMapping`. This is done using a custom `formatting_func`. However, tasks like Text Classification can also be defined using default definitions using the FeedbackDataset fields and questions. These mappings are then used for retrieving data from a dataset and initializing the training. We also offer some ideas for [unifying data](/guides/llms/practical_guides/collect_responses) out of the box.
+2. Next, we initialize the `ArgillaTrainer` and forward the task mapping and training framework. Internally, this uses the `FeedbackData.prepare_for_training`-method to format the data according to the expectations from the framework. Some other interesting methods are:
+   1. `ArgillaTrainer.update_config` to change framework specifc training paramaeters.
+   2. `ArgillaTrainer.train` to start training.
+   3. `ArgillTrainer.predict` to run inference.
+
+Underneath, you can see the happy flow for using the `ArgillaTrainer`.
```

Comment:
Happy flow?

Reply:
Yes, the easiest way something is likely to work.

Reply:
Haha awesome